mlir/include/mlir/Dialect/Rock/IR/RockAccelTuningParamAttrInterface.td

```diff
@@ -0,0 +1,110 @@
+//===- RockTuningParamAttrInterface.td - tuning params for the rock ops ---===//
+//
+// Part of the rocMLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2022 Advanced Micro Devices INc.
+//===----------------------------------------------------------------------===//
+//
+// This file defines RockAccelTuningParamAttrInterface, which abstracts provide
+// a common interface to handle tuning params for accelerators
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef ROCK_ACCEL_TUNING_PARAM_INTERFACE
+#define ROCK_ACCEL_TUNING_PARAM_INTERFACE
+
+include "mlir/Dialect/Rock/IR/RockBase.td"
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/OpBase.td"
+
+
+def RockAccelTuningParamAttrInterface : AttrInterface<"RockAccelTuningParamAttrInterface"> {
+  let description = [{
+    This interface is providing generic abstractions to handle tuning params for accelerators
+  }];
+  let cppNamespace = "::mlir::rock";
+
+  let methods = [
+    InterfaceMethod<
+        /*desc=*/[{
+          Return how many output rows each block computes. In other words, given that the input block
+          A layout is `[kOuter, mPerBlock, kpack]`, this method returns `mPerBlock`.
+        }],
+        /*retType=*/"int64_t",
+        /*methodName=*/"getMPerBlock",
+        /*args=*/(ins),
+        /*methodBody=*/"",
+        /*defaultImplementation=*/""
+      >,
+    InterfaceMethod<
+        /*desc=*/[{
+          Return how many output columns each block computes. In other words, given that the input block
+          A layout is `[kOuter, nPerBlock, kpack]`, this method returns `nPerBlock`.
+        }],
+        /*retType=*/"int64_t",
+        /*methodName=*/"getNPerBlock",
+        /*args=*/(ins),
+        /*methodBody=*/"",
+        /*defaultImplementation=*/""
+      >,
+    InterfaceMethod<
+        /*desc=*/[{
+          Given that the GEMM input blocks layout is `[kOuter, dPerBlock, kPack]` (where `d` can be `m` or `n`)
+          this method returns `kOuter`.
+        }],
+        /*retType=*/"int64_t",
+        /*methodName=*/"getKPerBlock",
```

Comment:
Since we're already renaming stuff, can we do ourselves a favor and call this `getKpacksPerBlock` or something? We might maybe clean up the main tuning parameters interface as well while we're here.